account_payment_date/models/payment_order.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# © <2018> <Cristian Salamea>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountPaymentMode(models.Model):
+    _inherit = 'account.payment.mode'
+
+    allow_force_date = fields.Boolean('Allow force accounting date')
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = 'account.payment.order'
+
+    def get_account_date(self):
```

Comment:
Why are you creating this function if after this you are also controlling if allow_force_date is checked or not, you could directly take the schedule date.